packages/zoe/zoe.chainmail

```diff
@@ -27,82 +28,66 @@ interface Zoe {
-   * represent proof of escrow and may be passed into a
-   * smart contract method.
-   */
-  getEscrowReceiptAssay() -> (Assay);
-
-  /**
-   * Returns the array of assays for the particular instance. This is 
-   * helpful in the case that the user has forgotten the order.
-   */
-  getAssaysForInstance() -> (List(Assay));
-
   /**
    * Create an installation by safely evaluating the code and
    * registering it with Zoe. Returns an installationHandle.
    */
-  install(code :String) -> Object
-
+  install(code :String, moduleFormat :String) -> (Object);
+  
   /** 
    * Zoe is long-lived. We can use Zoe to create smart contract
    * instances by specifying a particular contract installation to 
    * use, as well as the `terms` of the contract. The contract terms 
    * are the arguments to the contract, and must include
    * the expected assays for the underlying rights. (Other than the
    * `assays` property of `terms`, the `terms` properties are up to 
-   * the discretion of the smart contract.) We get back an instance, a
-   * handle for that instance, the handle for the installation, and the
-   * terms.
+   * the discretion of the smart contract.) We get back an invite (an
+   * ERTP payment) to participate in the contract.
    */
-  makeInstance(installationHandle :Object, terms :Object) -> (InstanceInformation);
+  makeInstance(installationHandle :Object, terms :Object) -> (Payment);
   
   /**
    * Credibly get the instance from the instanceHandle.
    */
   getInstance(instanceHandle :Object) -> (InstanceInformation);
 
   /** 
-   * To escrow, the user must provide a list of payments as well as
+   * To redeem an invite, the user must provide a list of payments as well as
    * their rules for the offer. 
    * 
    * The rules for the offer are in two parts: `payoutRules` are used 
    * by Zoe to enforce offer safety, and `exitRule` is used by Zoe 
    * to enforce exit safety. `payoutRules` is a list of objects, each 
-   * with a `kind` property (such as 'offerExactly') and a units
+   * with a `kind` property (such as 'offerAtMost') and a units
    * property. The objects in the `payoutRules` must be in the same order
    * as the assays associated with a smart contract. That is, the
    * units in index 0 of `payoutRules` should be a units for the assay
    * in index 0 of the assays array.`payments` is an array of the
    * actual payments to be escrowed, following the rules in the
-   * payoutRules. If the payoutRules kind is 'offerExactly' or 'offerAtMost',
+   * payoutRules. If the payoutRules kind is 'offerAtMost',
    * then a payment matching the units is expected. If the payoutRules
-   * kind is 'wantAtLeast' or 'wantExactly' then the payment will be
-   * ignored at escrow and should be `undefined`.
+   * kind is 'wantAtLeast' then the payment will be
+   * ignored and should be `undefined`.
    */
-  escrow (offerRules :OfferRules, payments :List(Payment)) -> (EscrowReceiptAndPayout);
```

Comment:
Is there updated literature for autoswap.md and public-auction.md? Both files still reference .escrow() and escrow receipts.

Reply:
You're right that we need to update all of the Zoe documentation including the contract explanations. Would you be able to help with that?

Reply:
Yeah, I can do that. I think what would be good is if you explained each change and deleted method (why it was changed/deleted, if anything replaces it and how it affects other parts of zoe). Then I can go through the .md files and update what I do understand, then get back to you with parts I don't.

Also, are the .js files also getting updated? Many of the examples in the .md files reference them.

List of methods in question, for easy reference:
- redeem (previously, escrow)
- makeInstance
- burnEscrowReceipt
- escrow
- getExtentsFor

List of changes in question
- escrow receipts
- payout kinds (was this just to simplify possible kinds?)
- "All offers to Zoe must redeem a Zoe invite" (what was it previously? why was it changed?)

Reply:
Also, should I remove the public-swap.md since the .js file is deleted?

Reply:
Sure, sounds like a good plan. 

> Also, are the .js files also getting updated? Many of the examples in the .md files reference them.

Oh, which ones do you mean? All of the changes in this PR will be merged into master so these updated contracts will be what our docs need to reference. 

> redeem (previously, escrow. Are all references to escrow(ing) not applicable anymore, or only in some cases? 

All references to the `escrow` function should be deleted. In its place, we have the `redeem` function which takes in an invite to a specific zoe contract, offerRules (a statement of what is offered and what is wanted) and the actual payments. So redeem is the escrowing step, although we aren't using that term for the method, if that makes sense.

> makeInstance
When a user calls `zoe.makeInstance` the contract makes a starting Zoe invite (in the form of an ERTP payment) which they get back. When the Zoe invite is redeemed, the user receives a `seat` object that has methods specific to that contract that they can call.

> burnEscrowReceipt

This sucker is gone! There are no escrow receipts anymore. 

> escrow

Gone.

> getExtentsFor

This is gone. We wanted to discourage use of extents by contract since they are the unlabeled amounts. Once we figured out how to have local version of the operations for units (UnitOps) we switched to that. Also, Mark pointed out that it was easier and clearer to get a full record of information about the offer and then get the units from it ( `zoe.getOffer(inviteHandle).units` ) so you will see that pattern in the contracts.

List of changes in question

> escrow receipts

These are all gone

> payout kinds (was this just to simplify possible kinds?)

Yeah, there are only two possible kinds now rather than four possible kinds. `wantExactly` and `haveExactly` got deleted. They weren't very useful and were cluttering up the code.

> "All offers to Zoe must redeem a Zoe invite" (what was it previously? why was it changed?)

Previously, if you wanted to make an offer, you would first escrow with Zoe and get an escrow receipt. One contract also created invites to use the contract (the covered call contract, where the "invite" is actually a financial option). We switched all the contracts to this style so we could get rid of the clunky escrow receipts.

> Also, should I remove public-swap.md since the .js file is deleted?

Public-swap got renamed to atomicSwap. We had to change the way it worked somewhat so now it's a basic atomic swap contract that is invite-only for making the matching offer.

Let me know if you have any more questions! I can hop on a call on Monday as well if that would help.

Reply:
> Oh, which ones do you mean? All of the changes in this PR will be merged into master so these updated contracts will be what our docs need to reference.

Oh, I think I was looking at the file before the PR changes. I clicked on a file and it showed me the one on master. 

> > redeem (previously, escrow. Are all references to escrow(ing) not applicable anymore, or only in some cases?
> 
> All references to the `escrow` function should be deleted. In its place, we have the `redeem` function which takes in an invite to a specific zoe contract, offerRules (a statement of what is offered and what is wanted) and the actual payments. So redeem is the escrowing step, although we aren't using that term for the method, if that makes sense.

Are we still using escrow as a term in general? 
For example, in the Zoe: Offer Safety Enforcement section, under the paragraph labeled "What is 'offer safety?'".
Also, under the description for zoe.redeem(): "...payments is an array of the actual payments to be escrowed..."

> Let me know if you have any more questions! I can hop on a call on Monday as well if that would help.

A call would be good, but I got a sore throat over the weekend so...probably not the best idea. :(

Reply:
> Oh, I think I was looking at the file before the PR changes. I clicked on a file and it showed me the one on master.

Gotcha, that makes sense.

> Are we still using escrow as a term in general?

Yeah, I think it makes sense to still use the term. There are a couple of things going on in the call to `redeem` but escrowing the payments that are sent in is one of those things.

> A call would be good, but I got a sore throat over the weekend so...probably not the best idea. :(

No problem, feel better! Let me know how I can help :)